mobius/src/geometry/line.rs

```diff
@@ -11,14 +50,40 @@ pub struct Line {
 impl Line {
     /// Create a line from a normal and distance offset
     /// This will automatically normalize the normal
-    pub fn new(normal: Complex, distance: f64) -> Result<Self, String> {
-        match normal.normalize() {
-            Some(unit_normal) => Ok(Self {
-                unit_normal,
-                distance,
-            }),
-            None => Err(String::from("Normal must be finite and non-zero")),
+    pub fn new(normal: Complex, distance: f64) -> Result<Self, LineError> {
+        ComplexError::require_finite_nonzero("normal", normal)?;
+        FloatError::require_finite("distance", distance)?;
```

Comment:
I don't like adding these checks all over the place, it feels more like validation code.

It might be nice to have `FiniteNonzero` or `Finite` wrapper around a Complex to restrict the types... though not sure I like the extra nesting and boilerplate required.